python/activator/middleware_interface.py

```diff
@@ -358,9 +383,10 @@ def prep_butler(self, visit: Visit) -> None:
         wcs = self._predict_wcs(detector, visit)
         center, radius = self._detector_bounding_circle(detector, wcs)
 
-        # central repo may have been modified by other MWI instances.
+        # repos may have been modified by other MWI instances.
         # TODO: get a proper synchronization API for Butler
         self.central_butler.registry.refresh()
+        self.butler.registry.refresh()
```

Comment:
I'm curious what this `refresh` and the one below are doing for you, especially after the switch to per-visit `MiddlewareInterface` objects.  Do workers ever need to use something that is only added to the repo by some other worker?  Or is this to avoid races when creating some shared repo content?

Reply:
Oh, sorry, I misread the original question. Yes, the `central_butler` refreshes are to avoid conflicts between concurrently writing workers. So far the only such conflicts have involved things like collection creation, and there shouldn't be any right now, but as with the local repo I'm trying to be robust to unknown future changes.